lib/api/objectHead.js

```diff
@@ -75,6 +77,18 @@ function objectHead(authInfo, request, log, callback) {
             if (headerValResult.error) {
                 return callback(headerValResult.error, corsHeaders);
             }
+            const partNumber = getPartNumber(request.query);
+            if (partNumber !== undefined) {
+                if (partNumber < 1 || partNumber > maximumAllowedPartCount) {
```

Comment:
`if partNumber > maximumAllowedPartCount` seems like a check we should have in a few other places too 👍